src/rewriters/Rewriter.ts

```diff
@@ -9,20 +9,32 @@ export interface RewriterOpts {
   fieldName?: string;
   rootTypes?: RootType[];
   matchConditions?: matchCondition[];
+  matchAnyPath?: boolean;
+  saveNode?: boolean;
 }
 
 /**
  * Abstract base Rewriter class
  * Extend this class and overwrite its methods to create a new rewriter
  */
 abstract class Rewriter {
+  public matchAnyPath: boolean = false;
+  public saveNode: boolean = false;
   protected rootTypes: RootType[] = ['query', 'mutation', 'fragment'];
   protected fieldName?: string;
   protected matchConditions?: matchCondition[];
 
-  constructor({ fieldName, rootTypes, matchConditions }: RewriterOpts) {
+  constructor({
+    fieldName,
+    rootTypes,
+    matchConditions,
+    matchAnyPath = false,
+    saveNode = false
```

Comment:
What do these 2 new variables do?

Reply:
The `matchAnyPath` is a non breaking change that allows users to trace all AST node paths not just fieldPaths, more thought for the CustomRewriter, since all of the existing ones are Field Specific.

The `saveNode` variable adds the ability to pass in the nodes and parents to the rewriteResponse function.

Reply:
Any reason not to always pass `nodeAndParents` to the `rewriteResponse` function? If it's just an extra param it shouldn't hurt anything?

Reply:
What do you think about naming it `matchNonFieldPaths` instead of `matchAnyPath` to make it more clear that the default is to match fields?

Reply:
- Im totally ok with removing `saveNode` and always store `nodeAndParents`.
- Im fine with hat too, although `matchNonFieldPaths` sounds like it comes without fieldPaths.

Reply:
That's a good point, what about `includeNonFieldPathsInMatch`?  It's long but should be more clear what's going on